### ISSUE TITLE
Layer masking (AKA Clip) a video using canvas?

### ISSUE THREAD

**Issue by Ivanca** [mifi/editly#221]
I want to create a circle using canvas and clip a video so only it uses the canvas element as alpha channel, how can I do that with editly?

The equivalent in Photoshop (and similarly on other software) looks a like this:
![image](https://user-images.githubusercontent.com/907138/213594251-44d0a2e2-a7ed-4db6-a20a-e2e0a01b3e45.png)

Thank you very much for this library.

**Comment by mifi** [comment]
Hi. I think what youre asking for is «masking». Not sure if canvas can do that

**Comment by Ivanca** [comment]
Yes it can, see https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/clip
But how can I do it with this library using a video?

**Comment by Ivanca** [comment]
I see that it can be done with FFMPEG as well and I assume that works much faster/cheaper, see https://curiosalon.github.io/blog/ffmpeg-alpha-masking/

Of course any of both ways would be welcomed, thanks.

**Comment by Ivanca** [comment]
I see that this project is already using webgl (on node) to do the transitions, it seems like this type of masking its feasible adding a little shader as explained here https://stackoverflow.com/questions/29307675/use-alpha-channeled-video-to-mask-html-content

**Comment by Ivanca** [comment]
I created a PR [https://github.com/mifi/editly/pull/222](https://github.com/mifi/editly/pull/222) that adds a new (optional) callback so users can use it to modify the generated image (_Fabric.Image_) before being rendered; so stuff like masking can be done using the fabric API with code like this:

```js
await editly({
  outPath: './alpha.mp4', 
  clips: [{
    duration: 4, layers: [
      { type: 'video', path: './assets/lofoten.mp4', cutFrom: 0, cutTo: 4 },
      { 
        type: 'video', 
        path: './assets/hiking.mp4',
        cutFrom: 0, cutTo: 4, 
        resizeMode: 'cover',
        originX: 'center', originY: 'center',
        left: 0.5, top: 0.5, width: 0.5, height: 0.5,
        fabricImagePostProcessing: async ({image, progress, canvas})=> {
          const circleArgs = {
            radius:  Math.min(image.width, image.height) * 0.4,
            originX: 'center',
            originY: 'center',
            stroke: 'white',
            strokeWidth: 22,
          };
          image.setOptions({
            clipPath: new fabric.Circle(circleArgs)
          });
          canvas.add(new fabric.Circle({
            ...circleArgs,
            left: image.getCenterPoint().x,
            top: image.getCenterPoint().y
          }));   
        }
      }
    ]}
  ]
});
```

Example result:


https://user-images.githubusercontent.com/907138/214545896-ab420beb-bd50-4974-9bad-9657e4f0c849.mp4

**Comment by Ivanca** [comment]
@mifi Any intention to merge this or a similar API in the near future? Thanks.

**Comment by bkeepers** [comment]
[0.15.0-rc.1](https://github.com/mifi/editly/releases/tag/v0.15.0-rc.1) was released today, which includes this new API. Check it out and let me know how it works for you.
